app/controllers/merchants_controller.rb

```diff
@@ -5,7 +5,7 @@ def index
   end
 
   def show
-    @merchant = Merchant.find(params[:id])
+    @merchant = User.find_by(id: session[:user_id])&.merchant || Merchant.find_by(id: params[:id])
```

Comment:
Either we get to this route by /merchants/:id or were getting the merchant from the logged in user its associated with

Reply:
OOoooo look at you with the memoization!